packages/core/components/other/nav-toggle/NavToggle.ts

```diff
@@ -1,162 +1,151 @@
-			return;
-		}
+	if (!nav) {
+		logger.error(`Could not find nav with selector: '${navSelector.value}'`);
+		return;
+	}
 
-		this.nav = nav;
+	bind();
 
-		this.bind();
-	}
+	function bind() {
+		on('click', clickHandler);
+		on('nav-toggle:show', onNavShow);
+		on('nav-toggle:hide', onNavHide);
 
-	bind() {
-		this.$on('nav-toggle:show', this.onNavShow);
-		this.$on('nav-toggle:hide', this.onNavHide);
-		this.$on('transitionstart', this.animStart, { target: this.nav });
-		this.$on('transitionend', this.animEnd, { target: this.nav });
+		if (nav) {
+			on('transitionstart', animStart, { target: nav });
+			on('transitionend', animEnd, { target: nav });
+		}
 	}
 
-	@bind('click')
-	onClick(e: Event) {
+	function clickHandler(e: Event) {
 		e.preventDefault();
 		e.stopPropagation();
 
-		if (this.isOpen) {
-			this.hide(this.hideImmediately);
+		if (isOpen) {
+			hide(hideImmediately.value);
 		} else {
-			this.show(this.showImmediately);
+			show(showImmediately.value);
 		}
 	}
 
-	onNavShow(e: toggleEvent) {
-		this.show(e.detail);
+	function onNavShow(e: toggleEvent) {
+		show(e.detail);
 	}
 
-	onNavHide(e: toggleEvent) {
-		this.hide(e.detail);
+	function onNavHide(e: toggleEvent) {
+		hide(e.detail);
 	}
 
-	animStart(e: Event) {
-		if (e.target !== this.nav) {
+	function animStart(e: Event) {
+		if (e.target !== nav) {
 			return;
 		}
 
-		if (this.animStarted) {
+		if (animStarted) {
 			return;
 		}
 
-		this.animStarted = true;
+		animStarted = true;
 
-		if (this.isOpen) {
-			this.html.classList.add(`${this.navName}-anim`);
+		if (isOpen) {
+			html.classList.add(`${navName.value}-anim`);
 		} else {
-			this.html.classList.add(`${this.navName}-hide-anim`);
+			html.classList.add(`${navName.value}-hide-anim`);
 		}
 	}
 
-	animEnd(e: Event) {
-		if (e.target !== this.nav) {
+	function animEnd(e: Event) {
+		if (e.target !== nav) {
 			return;
 		}
 
-		if (!this.animStarted) {
+		if (!animStarted) {
 			return;
 		}
 
-		this.animStarted = false;
+		animStarted = false;
 
-		if (this.isOpen) {
-			this.html.classList.remove(`${this.navName}-anim`);
+		if (isOpen) {
+			html.classList.remove(`${navName.value}-anim`);
 		} else {
-			this.html.classList.remove(`${this.navName}-hide-anim`);
+			html.classList.remove(`${navName.value}-hide-anim`);
 		}
 	}
 
-	show(immediately = false) {
-		this.isOpen = true;
+	function show(immediately = false) {
+		isOpen = true;
 
 		if (immediately) {
-			this.html.classList.add(`${this.navName}-show-immediately`);
+			html.classList.add(`${navName.value}-show-immediately`);
 
 			requestAnimationFrame(() => {
-				this.html.classList.remove(`${this.navName}-show-immediately`);
+				html.classList.remove(`${navName.value}-show-immediately`);
 			});
 		}
 
-		this.html.classList.add(`${this.navName}-visible`);
-		this.$element.setAttribute('aria-expanded', 'true');
+		html.classList.add(`${navName.value}-visible`);
+		nav?.setAttribute('aria-expanded', 'true');
```

Comment:
nope, tutaj był `this.$element`, wiec będzie `element` z `defineComponent`, które teraz masz jako `_`